consensus/src/consensus/consensus_proxy.rs

```diff
@@ -210,10 +222,13 @@ impl<N: Network> ConsensusProxy<N> {
         // We drop the blockchain lock because it's no longer needed while we request proofs
         drop(blockchain);
 
-        if receipts
-            .iter()
-            .any(|(_, block_number)| block_number > &current_block_number)
-        {
+        if receipts.iter().any(|(_, block_number)| {
+            if let Some(block_number) = block_number {
+                block_number > &current_block_number
+            } else {
+                false
+            }
```

Comment:
Can maybe be shortened to `block_number.unwrap_or(0)`:

```suggestion
            block_number.unwrap_or(0) > &current_block_number
```